pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -513,6 +513,7 @@ func (cr *PerconaXtraDBCluster) CheckNSetDefaults(serverVersion *ServerVersion)
 	}
 
 	cr.setSecurityContext()
+	cr.Spec.Platform = serverVersion.Platform
```

Comment:
It's already was set a few lines before.

Reply:
8578c8d76d9a1711eabd53fe7fcf4be3f840e62b